src/modal/modal.js

```diff
@@ -358,9 +360,26 @@ angular.module('ui.bootstrap.modal', ['ui.bootstrap.transition'])
             var templateAndResolvePromise =
               $q.all([getTemplatePromise(modalOptions)].concat(getResolvePromises(modalOptions.resolve)));
```

Comment:
I've reorganized things so that the opened promise is the only one burdened with chaining concerns.  It shuffles things around a little, but it leaves the `templateAndResolvePromise` alone.